app/assets/javascripts/Components/peer_reviews_manager.jsx

```diff
@@ -0,0 +1,334 @@
+  };
+
+  performButtonAction = (action) => {
+    let reviewees = this.studentsTable ? this.studentsTable.state.selection : [];
+    let reviewers = this.ReviewersTable ? this.ReviewersTable.state.selection : [];
+
+    if ((action == 'assign' || action == 'random_assign') && reviewers.length === 0) {
+      alert('No reviewers selected.');
+      return;
+    }
+
+    if ((action == 'assign' || action == 'random_assign') && reviewees.length === 0) {
+      alert('No reviewees selected.');
+      return;
+    }
+
+    this.setState({loading: true});
+    $.post({
+      url: Routes.assign_groups_assignment_peer_reviews_path(this.props.assignment_id),
+      data: {
+        actionString: action,
+        selectedReviewerGroupIds: reviewers,
+        selectedRevieweeGroupIds: reviewees,
+        selectedReviewerInRevieweeGroups: this.state.selectedReviewerInRevieweeGroups,
+        numGroupsToAssign: this.state.numReviewers
+      }
+    }).then(this.fetchData);
+  };
+
+  render(){
+    return (
+      <div>
+        <GradersActionBox
+          performAction={this.performButtonAction}
+          updateNumReviewers={this.updateNumReviewers}
+        />
+        <div className='mapping-tables'>
+          <div className='mapping-table'>
+            <ReviewersTable
+              ref={(r) => this.ReviewersTable = r}
+              groups={this.state.reviewerGroups}
+              reviewerToNumReviews={this.state.reviewerToNumReviews}
+              loading={this.state.loading} />
+          </div>
+          <div className='mapping-table'>
+            <RevieweesTable
+              ref={(r) => this.studentsTable = r}
+              groups={this.state.revieweeGroups}
+              revieweeToReviewers={this.state.revieweeToReviewers}
+              groupIdToName={this.state.groupIdToName}
+              onReviewerChangeInRevieweeTable={this.updatedSelectedReviewersInRevieweesTable}
+              selectedReviewerInRevieweeGroups={this.state.selectedReviewerInRevieweeGroups}
+              loading={this.state.loading}
+              showSections={this.props.showSections}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+
+class RawReviewersTable extends React.Component {
+  static columns = [
+    {
+      show: false,
+      accessor: '_id',
+      id: '_id'
+    },
+    {
+      Header: 'Reviewer Groups',
+      accessor: 'name',
+      id: 'name'
+    },
+    {
+      Header: 'Num Reviews',
+      accessor: 'groups',
+      className: 'groups',
+      filterable: false
+    }
+  ];
+
+  clearCheckboxes = () => {
+    this.refs.table.clearCheckboxes();
+  };
+
+  changeSection = (event) => {
+    this.clearCheckboxes();
+    this.setState({ sectionName: event.target.value });
+    this.props.refresh();
+  };
+
+  render() {
+    const hashmap = this.props.reviewerToNumReviews;
+    const groups_data = this.props.groups.map(function(group) {
```

Comment:
Use the newer arrow functions `(group) => { ...}`. This way you don't need to bind `this.